### PR TITLE
Migrate Subiquity client to null-safety

### PR DESCRIPTION
**Pull request by jpnurmi**
The migration was done with help of the migration tool: https://dart.dev/null-safety/migration-guide. The `.g.dart` and `.freezed.dart` files were re-generated with the latest `freezed` (that no longer supports non-null-safe code).

### CODE REVIEW

**Comment by jpnurmi** [comment]
Hey, I've been struggling a bit with dependencies, particularly in `ubuntu_desktop_installer`. I wanted to use [pattern matching](https://pub.dev/packages/freezed#unionssealed-classes) for #30, but there's a tricky conflict with the current dependencies :

```
Bad state: Unexpected diagnostics:
/home/jpnurmi/.fvm/versions/dev/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:166:25 - This requires the 'non-nullable' language feature to be enabled.
/home/jpnurmi/.fvm/versions/dev/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:30:49 - This requires the 'non-nullable' language feature to be enabled.
/home/jpnurmi/.fvm/versions/dev/bin/cache/pkg/sky_engine/lib/ui/channel_buffers.dart:390:19 - This requires the 'non-nullable' language feature to be enabled.
[...]
```

Migrating to null-safety would help to upgrade all dependencies to the latest versions, which would solve the conflict too. I have already verified that migrating `ubuntu_desktop_installer` is no problem once `subiquity_client` has been migrated.

**Review by oSoMoN — CHANGES_REQUESTED** [review]
Would you mind extracting [this commit](682b267b47e04d1879ecfac74a7468f193c759ba) into a separate PR?

**Comment by jpnurmi** [comment]
Done: https://github.com/canonical/ubuntu-desktop-installer/pull/74. Should've done it in the first place but I was lazy. :)

File: packages/subiquity_client/pubspec.yaml [hunk 1]
```diff
@@ -4,19 +4,17 @@ version: 0.0.1
...
-  effective_dart: ^1.3.0
-  build_runner: ^1.11.5
-  freezed: ^0.12.7
-  json_serializable: ^3.5.1
-  test: ^1.16.0
+  build_runner: ^2.0.1
+  effective_dart: ^1.3.1
+  freezed: ^0.14.1
+  json_serializable: ^4.1.1
+  test: ^1.17.0
```
Comment (jpnurmi) [inline packages/subiquity_client/pubspec.yaml:20]: Sorry for the re-ordering here. I'm using the Pubspec Assistant VS Code extension for managing dependencies. It likes to keep these in alphabetical order - I hope that's ok...

File: packages/subiquity_client/pubspec.yaml [hunk 2]
```diff
@@ -4,19 +4,17 @@ version: 0.0.1
 homepage: https://github.com/canonical/ubuntu-desktop-installer
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.12.0 <3.0.0'
```
Comment (jpnurmi) [inline packages/subiquity_client/pubspec.yaml:7]: (not that it matters much, but the migration tool prefers single quotes)

**Comment by oSoMoN** [comment]
LGTM, thanks!

**Pull request merged**